### ISSUE TITLE
Feature request: File transfer and data streaming

### ISSUE THREAD

**Issue by kap2fox** [algorealmInc/SwarmNL#38]
We have a scenario with data transfer in local network. And we need two types of data transferring - the whole file with guarantee delivery; the data streaming when we don't care about some messages.
We'd like to have functiron like
_node.send_data(to: PeerId,data: bytes)->Result// send data to another node_

and

_node.wait_for_data()->Result<DataFromAnotherNode{ // accept data from another nodes origin_from: PeerId, data: bytes}>_

**Comment by thewoodfish** [comment]
Please can you elaborate on the "data streaming" part?

**Comment by thewoodfish** [comment]
For the file transfer, there is a maximum amount of bytes that can be sent via RPC or gossiped. If the file is larger, it will most likely be broken to fit and then reassembled on the receiving end. 
Also, there has to be some mechanism to ensure the file is correct, maybe after transfer of each byte chuck a hash can be sent back to confirm the authenticity and completeness of the data sent. That should work

**Comment by kap2fox** [comment]
> Please can you elaborate on the "data streaming" part?

I'm talking about data sequences delivering without guaranteed of integrity. The part of the data may be lost but important to get it periodically for monitoring task.

**Comment by thewoodfish** [comment]
Yes, Swarm makes provisions for that too. You can even multicast if you desire.

**Comment by sacha-l** [comment]
Hi @kap2fox it's been a while! Not sure if you tried out the library since it's last release? Happy to step you through it!
